src/commands/Checkin.ts

```diff
@@ -68,6 +71,15 @@ export default class Checkin extends Command {
     // Get arguments. Get rid of the null types by checking them.
     const publicArgument = interaction.options.getBoolean('public');
     const widescreenArgument = interaction.options.getBoolean('widescreen');
+    const dateArgument = interaction.options.getString('date');
+
+    // Regex to match dates in the format of MM/DD(/YYYY) or MM-DD(-YYYY).
+    const regexp = new RegExp('^(\\d{1,2})(/|-)(\\d{1,2})((/|-)(\\d{2}|\\d{4})){0,1}$', 'g');
+    const dateMatches = regexp.exec(dateArgument!);
+
+    const month = dateMatches?.[1] ? parseInt(dateMatches[1], 10) : DateTime.now().month;
+    const day = dateMatches?.[3] ? parseInt(dateMatches[3], 10) : DateTime.now().day;
+    const year = dateMatches?.[6] ? parseInt(dateMatches[6], 10) : DateTime.now().year;
```

Comment:
Do we have error messaging if the date is invalid? i.e. send a response if they put 13/32/2023